selectors/views/custom_status.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {getCurrentUser, getUser} from 'mattermost-redux/selectors/entities/users';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+import {Preferences} from 'mattermost-redux/constants';
+
+import {GlobalState} from 'types/store';
+
+export function getCustomStatus(state: GlobalState, userID?: string) {
+    const user = userID ? getUser(state, userID) : getCurrentUser(state);
+    const userProps = user.props || {};
+    return userProps.customStatus ? JSON.parse(userProps.customStatus) : {};
```

Comment:
Also, returning an empty object literal can cause similar issues where a new object is returned each time its called for a user that doesn't have a custom status set. You used `defaultProps` in a lot of places to set a default object for custom statuses, so perhaps we could just return undefined here?

If you get an ESList error from doing that, feel free to add an eslint-ignore-line for that. The `no-undefined` rule, if it's enabled, is something we've wanted to look into turning off for a while now.